src/main/scala/org/apache/spark/deploy/armada/Config.scala

```diff
@@ -120,4 +133,79 @@ private[spark] object Config {
       .stringConf
       .checkValue(serviceNamePrefixValidator, "Service name prefix must adhere to rfc 1035 label names.")
       .createWithDefaultString("armada-spark-driver-")
+
+  private val invalidLabelListErrorMessage =
+    "Must be a comma-separated list of valid Kubernetes labels (each as key=value)"
+
+  val ARMADA_SPARK_GLOBAL_LABELS: ConfigEntry[String] =
+    ConfigBuilder("spark.armada.global.labels")
+      .doc("A comma separated list of kubernetes labels (in key=value format) to be added to all " +
+        "both the driver and executor pods.")
+      .stringConf
+      .checkValue(selectorsValidator, invalidLabelListErrorMessage)
+      .createWithDefaultString("")
+
+  val ARMADA_SPARK_DRIVER_LABELS: ConfigEntry[String] =
+    ConfigBuilder("spark.armada.driver.labels")
+      .doc("A comma separated list of kubernetes labels (in key=value format) to be added to the " +
+        "driver pod.")
+      .stringConf
+      .checkValue(k8sLabelListValidator, invalidLabelListErrorMessage)
+      .createWithDefaultString("")
+
+  val ARMADA_SPARK_EXECUTOR_LABELS: ConfigEntry[String] =
+    ConfigBuilder("spark.armada.executor.labels")
+      .doc("A comma separated list of kubernetes labels (in key=value format) to be added to all " +
+        "executor pods.")
+      .stringConf
+      .checkValue(k8sLabelListValidator, invalidLabelListErrorMessage)
+      .createWithDefaultString("")
+
+  private def k8sLabelListValidator(labelList: String): Boolean = {
+    // empty is OK
+    val s = labelList.trim
+    if (s.isEmpty) true
+    else {
+      // every entry must be key=value, and both key & value pass the validator
+      s.split(",").forall { entry =>
+        entry.split("=", 2) match {
+          case Array(key, value) =>
+            K8sLabelValidator.isValidKey(key) && K8sLabelValidator.isValidValue(value)
+          case _ =>
+            false
+        }
+      }
+    }
+  }
 }
+
+object K8sLabelValidator {
+  // DNS1123 label: 1–63 chars, alphanumeric start/end, interior may have '-'
+  private val DNS1123_LABEL =
+    "([A-Za-z0-9]([A-Za-z0-9\\-]*[A-Za-z0-9])?)"
```

Comment:
Does that middle chunk (`[A-Za-z0-9\\-]*`) need to also accept periods (`.`) and underscores (`_`)?  Reading https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/ , I'm not sure if they specify that for all labels or if DNS-1123 labels are more restrictive?